generators/go-v2/sdk/src/wire-tests/WireTestGenerator.ts

```diff
@@ -119,13 +119,27 @@ export class WireTestGenerator {
         imports.set("bytes", "bytes");
         imports.set("encoding/json", "encoding/json");
 
+        // Track test function names to avoid duplicates (multiple endpoints can map to the same test name)
+        const seenTestFunctionNames = new Set<string>();
+
         const endpointTestCaseCodeBlocks = endpoints
             .map((endpoint) => {
                 const snippet = endpointTestCases.get(endpoint.id);
                 if (!snippet) {
                     this.context.logger.warn(`No snippet found for endpoint ${endpoint.id}`);
                     return null;
                 }
+
+                // Parse the test function name from the snippet and skip if we've already seen it
+                const testFunctionName = this.parseTestFunctionNameFromSnippet(snippet);
```

Comment:
let's include it but deduplicate the name. Can we keep a counter and add it? Like if we see duplicates, add Test1, Test2, Test3 ...

Also very important, this MUST be deterministic. Which should be okay if everything we are iterating over is ordered